Refactor RealPath (#198)
laminas/laminas-filter: docs/book/v3/standard-filters.md

### Basic Usage

For any given link or pathname, its absolute path will be returned.
References to `/./`, `/../` and extra `/` sequences in the input path will be stripped.
The resulting path will not have any symbolic links, `/./`, or `/../` sequences.

> `Laminas\Filter\RealPath` will return `FALSE` on failure, e.g. if the file does not exist.

Reviewer: ```suggestion
`Laminas\Filter\RealPath` will return the un-filtered input on failure, for example, if the input is not a string, the value cannot be interpreted as a file path, or, the path does not exist when existence is a configured requirement.
```

> On BSD systems `Laminas\Filter\RealPath` doesn't fail if only the last path component doesn't exist, while other systems will return `false`.

Reviewer: Can we verify this? also, it shouldn't return false.

### Non-Existing Paths

> Sometimes it is useful to get paths to files that do n0t exist; e.g., when you

Reviewer: ```suggestion
Sometimes it is useful to get paths to files that do not exist; e.g., when you
```

want to get the real path for a path you want to create. You can then

> provide a `FALSE` `exists` option at initiation.

Reviewer: ```suggestion
provide `false` for the `exists` option during construction.
```

Reviewer: Also, whilst the current docs don't conform to this rule - can you please re-format _(just the changes)_ to one sentence per line please. This is a general, un-documented docs style preference, that hopefully we'll add to a style guide one day 😬

Reviewer 2: > # Writing Style
> 
> ## One Sentence per-line Method
> 
> Documentation should be written using [the one sentence per-line method](https://asciidoctor.org/docs/asciidoc-recommended-practices/#one-sentence-per-line). 
> There are a number of excellent reasons for writing in this style, including:
> 
> - It prevents reflows (meaning a change early in the paragraph won’t cause the remaining lines in the paragraph to reposition)
> - You can easily swap/move sentences
> - You can easily separate or join paragraphs
> - You can comment out sentences or add commentary to them
> - You can spot sentences which are too long or sentences that vary widely in length
> - You can spot redundant (and thus mundane) patterns in your writing
> 
> Take the following content as an example.
> 
> ```markdown
> The concrete `TableGateway` object uses constructor injection for getting
> dependencies and options into the instance. The table name and an instance of
> an `Adapter` are all that is required to create an instance. Out of the box, this 
> implementation makes no assumptions about table structure or metadata, and 
> when `select()` is executed, a simple `ResultSet` object with the populated 
> `Adapter`'s `Result` (the datasource) will be returned and ready for iteration.
> ```
> 
> While it can be easily read in any editor, as it does not require the reader to scroll, that simplicity comes at the cost of maintainability. 
> For example, if you needed to put the second sentence before the first, or move it to any other location, it would need to be cut out of the surrounding paragraph before it could be moved.
> 
> Compare that with the following sample content.
> 
> ```markdown
> The concrete `TableGateway` object uses constructor injection for getting dependencies and options into the instance. 
> The table name and an instance of an `Adapter` are all that is required to create an instance.
> Out of the box, this implementation makes no assumptions about table structure or metadata, and when `select()` is executed, a simple `ResultSet` object with the populated `Adapter`'s `Result` (the datasource) will be returned and ready for iteration.
> ```
> 
> Yes, the content, by default scrolls out of view. 
> However, most text editors and IDEs support "_soft wrapping_", making this a non-issue. 
> Using this approach, the sentence doesn't need to be extracted from the surrounding paragraph before it can be moved.

Reviewer: Sorry Frank! the _documented_ style guide… Where is it?

Author: Will do. Thanks both
